src/Glpi/Form/Migration/MigrationManager.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\Migration;
+
+use DBmysql;
+use Glpi\Console\AbstractCommand;
+use Glpi\Console\Migration\FormCreatorPluginToCoreCommand;
+use Plugin;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class MigrationManager
+{
+    private DBmysql $DB;
+    private ?AbstractCommand $command;
+    private array $keysMap = [];
+
+    public function __construct(DBmysql $DB, ?AbstractCommand $command = null)
+    {
+        $this->DB      = $DB;
+        $this->command = $command;
```

Comment:
I am not fan a linking this object to the command that invoked it.

I understand that it make it easier to output errors but it only works in the context of the command.
Later, if we want to execute this action from the UI we won't be able to get the errors.

A more solid approach is to let the `doMigration()` method return some kind of result object like a `FormMigrationResult`.
This allow you to store any error you want into this object and the general success/failure state.
Then, no matter where you need to handle the results (UI, command, tests, ...) you can read the errors from the object and display them accordingly.